lib/property_sets/property_set_model.rb

```diff
@@ -99,12 +99,16 @@ def keys
         @properties.keys
       end
 
-      def default(key)
-        PropertySets::Casting.read(type(key), raw_default(key))
+      def default(key, owner=nil)
+        PropertySets::Casting.read(type(key), raw_default(key, owner))
       end
 
-      def raw_default(key)
-        @properties[key].try(:[], :default)
+      def raw_default(key, owner=nil)
+        default_value = @properties[key].try(:[], :default)
+        if default_value.is_a?(Proc)
```

Comment:
`respond_to? :call` might be nice to be able to set a class or something like that